src/main/java/pixelitor/selection/Selection.java

```diff
@@ -370,6 +370,17 @@ public DebugNode createDebugNode(String name) {
         return node;
     }
 
+    public void update(SelectionType selectionType, Object mouseInfo) {
```

Comment:
The updateInProgressSelection method in SelectionBuilder could certainly be improved, but I don't see this extraction as a good first step, because the main problem I see is the unclear semantics of the mouseInfo argument, and this problem is not adressed at all.